composer.json

```
@@ -10,12 +10,37 @@
     }
   },
   "require-dev": {
-    "php": "^7.4|^8.0",
+    "php": "^8.0",
     "phpunit/phpunit": "^9.6",
     "slevomat/coding-standard": "^8.14.1",
     "squizlabs/php_codesniffer": "^3.9"
   },
   "scripts": {
+    "post-install-cmd": [
+        "@tool:configlet:fetch"
+    ],
+    "post-update-cmd": [
+        "@tool:configlet:update"
+    ],
+    "ci": [
+        "@configlet:fmt",
+        "@lint:check",
+        "@tests:run"
+    ],
+    "tool:configlet:error": "[ -x bin/configlet ] || ( echo 'Fetch configlet first! Run: \"bin/fetch-configlet\"'; false )",
+    "tool:configlet:fetch": "[ -z \"$CI\" ] || [ -x bin/configlet ] || bin/fetch-configlet",
+    "tool:configlet:update": [
+        "rm bin/configlet || true",
+        "[ -n \"$CI\" ] || bin/fetch-configlet"
+    ],
+    "configlet:fmt": [
+        "@tool:configlet:error",
```

Does it make sense to just fetch it here ?

```suggestion
        "@tool:configlet:fetch",
```

---

This is a point, where I am not 100% sure. I usually argue, that when you reach this point, you haven't used `composer install` or `... update`, so I see it as an error. Mostly because I cannot rely on the development environment in total. And very often the attempt to fix that causes more problems than it solves.

On the other hand, sometimes this might be enough. But still, when you get there, you did something that was not planned for the automation to do. Must I support this? I don't know...

---

I chose to simply do nothing in case of a missing configlet. The command will just fail for "command not found".